atlp-rwanda/cabal-phantom-backend
# #AO8d4STV Assign buses to routes

### What does this PR do?
It requests to merge ft-assign-buses-to-routes-AO8d4STV into develop
###  Description of Task to be completed?

- Assign registered buses to their routes

- Unassign assigned buses from routes

- See a paginated list of users

###  How should this be manually tested?
You may use this **https://phantom-caba-ft-assign--l3opgl.herokuapp.com/** as the host to perform tasks below **OR**
- Clone the repo
- Install all packages
- Set up dotenv with .env.example as reference
- Run this command npm run migrateDB to make migration and seeding

- Login using {
"email":"admin@phantom.com",
"password":"admin"
}

- To access these end point PATCH **/api/v1/buses/assignroutes/{id of bus}** to assign
then use PATCH **/api/v1/buses/unassignroutes/{id of bus}**  you have to login with these credentials:
in body use `"routeID":x` where x is an integer
GET /api/v1/buses/routes/assignedroute?page=1&limit=4 TO see a paginated list of assigned buses to routes
###  What are the relevant pivotal tracker stories?
[AO8d4STV](https://trello.com/c/AO8d4STV/13-assign-buses-to-routes)
###  Screenshots (if appropriate)
 
<img width="1440" alt="Screen Shot 2020-11-09 at 17 16 27" src="https://user-images.githubusercontent.com/68019992/98559515-612ea480-22af-11eb-9a3d-d56f6e6c78df.png">
